bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -452,6 +402,99 @@ public String toString() {
 
     // Deprecated methods
 
+    /**
+     * <p>Creates an ObjectId using time, machine and inc values.  The Java driver used to create all ObjectIds this way, but it does not
+     * match the <a href="http://docs.mongodb.org/manual/reference/object-id/">ObjectId specification</a>, which requires four values, not
+     * three. This major release of the Java driver conforms to the specification, but still supports clients that are relying on the
+     * behavior of the previous major release by providing this explicit factory method that takes three parameters instead of four.</p>
+     *
+     * <p>Ordinary users of the driver will not need this method.  It's only for those that have written there own BSON decoders.</p>
+     *
+     * <p>NOTE: This will not break any application that use ObjectIds.  The 12-byte representation will be round-trippable from old to new
+     * driver releases.</p>
+     *
+     * @param time    time in seconds
+     * @param machine machine ID
+     * @param inc     incremental value
+     * @return a new {@code ObjectId} created from the given values
+     * @since 2.12.0
+     * @deprecated Use {@link #ObjectId(int, int)} instead
+     */
+    @Deprecated
+    public static ObjectId createFromLegacyFormat(final int time, final int machine, final int inc) {
+        return new ObjectId(time, machine, inc);
+    }
+
+    /**
+     * Gets the current value of the auto-incrementing counter.
+     *
+     * @return the current counter value.
+     * @deprecated
+     */
+    @Deprecated
+    public static int getCurrentCounter() {
+        return NEXT_COUNTER.get();
+    }
+
+    /**
+     * Gets the generated machine identifier.
+     *
+     * @return an int representing the machine identifier
+     * @deprecated
+     */
```

Comment:
We no longer need to generate a machine ID with the new format. What would be the proper value to return for this method?